modules/contact/src/auxkernels/MortarFrictionalStateAux.C

```diff
@@ -0,0 +1,85 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MortarFrictionalStateAux.h"
+#include "SystemBase.h"
+
+registerMooseObject("ContactApp", MortarFrictionalStateAux);
+
+InputParameters
+MortarFrictionalStateAux::validParams()
+{
+  InputParameters params = AuxKernel::validParams();
+  params.set<ExecFlagEnum>("execute_on") = EXEC_NONLINEAR;
+  params.addClassDescription("This class creates discrete states for nodes into frictional "
+                             "contact, including contact/no-contact and stick/slip.");
+  params.addRequiredCoupledVar("tangent_one",
+                               "First tangent vector Lagrange multiplier for computing the mortar "
+                               "frictional pressure vector.");
+  params.addRequiredCoupledVar("tangent_two",
+                               "Second tangent vector Lagrange multiplier for computing the mortar "
+                               "frictional pressure vector.");
+  params.addRequiredCoupledVar(
+      "contact_pressure",
+      "Normal contact pressure Lagrange multiplier from the mortar contact enforcement.");
+  params.addRequiredParam<Real>("mu", "Friction coefficient to compute nodal states");
+  params.addParam<Real>("tolerance", 1.0e-3, "Tolerance value used to determine the states");
+  params.addParam<bool>(
+      "use_displaced_mesh", true, "Whether to use the displaced mesh to get the mortar interface.");
+  return params;
+}
+
+MortarFrictionalStateAux::MortarFrictionalStateAux(const InputParameters & params)
+  : AuxKernel(params),
+    _tangent_one(&coupledValueLower("tangent_one")),
+    _tangent_two(&coupledValueLower("tangent_two")),
+    _contact_pressure(&coupledValueLower("contact_pressure")),
```

Comment:
```suggestion
    _tangent_one(coupledValueLower("tangent_one")),
    _tangent_two(coupledValueLower("tangent_two")),
    _contact_pressure(coupledValueLower("contact_pressure")),
```

Reply:
I enabling use of the mortar aux kernel by 2D problems. For that reason, I am keeping `tangent_two` as a pointer, but maybe there was a better way to do that.